components/nexus-repository/src/main/java/org/sonatype/nexus/repository/manager/RepositoryManagerImpl.java

```diff
@@ -146,8 +150,14 @@ private void untrack(final Repository repository) {
   protected void doStart() throws Exception {
     List<Configuration> configurations = store.list();
     if (configurations.isEmpty()) {
-      log.debug("No repositories configured");
-      return;
+      log.debug("No repositories configured, installing defaults");
+      for (InitialRepositoryConfiguration configProvider : initialRepositoryConfigurations) {
+        for (Configuration configuration : configProvider.getRepositoryConfigurations()) {
+          log.debug("Provisioning default repository: {}", configuration);
+          store.create(configuration);
+        }
+      }
+      configurations = store.list();
```

Comment:
configurations could still be empty if no InitialRepositoryConfiguration or they returned empty lists.  Edge case but that would then show odd logging below of "Restoring 0 repositories" etc